operations/addOperation.h

```diff
@@ -11,10 +11,8 @@ two tensors.
 template <typename T>
 class AddOperation : public Operation<T> {
     public:
-    Tensor<T> *t1 = NULL, *t2 = NULL; // Input tensors
-    Tensor<T> *t3 = NULL; // Output tensor
-
-    AddOperation(Tensor<T> *t1, Tensor<T> *t2) : Operation<T>(t1,t2){
+    
+    AddOperation(Tensor<T> *t1, Tensor<T> *t2) {
```

Comment:
This is  because I've removed the t1 ,t2 and t3 declarations from the inherited class. now whenever I refer to t1,t2 and t3. They refer to the variables in the base Operation class. Hence, there is no need for the Operation(t1,t2) call of the base class to set them as they are already set.